sdk/src/common/global_log_handler.cc

```diff
@@ -0,0 +1,71 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "opentelemetry/sdk/common/global_log_handler.h"
+
+#include <cstring>
+#include <random>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace common
+{
+namespace internal_log
+{
+
+LogHandler::~LogHandler() {}
+
+void DefaultLogHandler::Handle(LogLevel level,
+                               const char *file,
+                               int line,
+                               const char *msg,
+                               const sdk::common::AttributeMap &attributes) noexcept
+{
+  std::stringstream output_s;
+  output_s << "[" << LevelToString(level) << "] ";
+  if (file != nullptr)
+  {
+    output_s << "File: " << file << ":" << line;
+  }
+  if (msg != nullptr)
+  {
+    output_s << msg;
+  }
+  output_s << std::endl;
+  // TBD - print attributes
+  std::cout << output_s.str();  // thread safe.
+}
+
+void NoopLogHandler::Handle(LogLevel level,
+                            const char *file,
+                            int line,
+                            const char *msg,
+                            const sdk::common::AttributeMap &attributes) noexcept
+{
+  std::stringstream output_s;
+  output_s << "[" << LevelToString(level) << "] ";
+  if (file != nullptr)
+  {
+    output_s << "File: " << file << ":" << line;
+  }
+  if (msg != nullptr)
+  {
+    output_s << msg;
+  }
+  output_s << std::endl;
+  // TBD - print attributes
+  std::cout << output_s.str();  // thread safe.
+}
+
+std::pair<nostd::shared_ptr<LogHandler>, LogLevel> &GlobalLogHandler::GetHandlerAndLevel() noexcept
+{
+  static std::pair<nostd::shared_ptr<LogHandler>, LogLevel> handler_and_level{
+      nostd::shared_ptr<LogHandler>(new DefaultLogHandler), LogLevel::Warning};
```

Comment:
The singleton is in the *.cc file now, great.

As a side note, I reviewed every occurrence of the `static` keyword in every header file under sdk/include.

There are a few other static variables used, but none of them are required to be unique, so the scope of the fix is really narrowed down to this one alone, `handler_and_level` in `GlobalLogHandler::GetHandlerAndLevel()`.

Reply:
Yes, we can raise another issue to move all static variables into `*.cc`, it's still a waste of memory.

Reply:
Agree, filed issue #1429 then.